File: stack/stack.ts
import { DoublyLinkedList } from "../doubly_linked_list/doublyLinkedList";

export class Stack {
  constructor(
    public length: number = 0,
    public storage = new DoublyLinkedList()
  ) {}

  push(value) {
    this.storage.addToHead(value);
    this.length++;
  }

  pop() {
    if (this.length === 0) {
      return null;
    } else {
      this.length--;
      return this.storage.removeFromHead();
    }
  }
}

File: data_structures/stack/stack.py
import sys
sys.path.append('../doubly_linked_list')
from doubly_linked_list import DoublyLinkedList, ListNode

class Stack:
    def __init__(self):
        self.size = 0
        self.storage = DoublyLinkedList()

    def __len__(self):
        return self.size

    def push(self, value):
        self.storage.add_to_tail(value)
        self.size += 1

    def pop(self):
        # handle if stack if empty
        if self.size == 0:
            return None

        self.size -= 1
        return self.storage.remove_from_tail()
